src/audio/codec_adapter/codec_adapter.c

```diff
@@ -384,13 +384,14 @@ int codec_adapter_copy(struct comp_dev *dev)
 		codec_adapter_copy_from_source_to_lib(&source->stream, &codec->cpd,
 						      codec_buff_size);
 		codec->cpd.avail = codec_buff_size;
-		ret = codec_init_process(dev);
+		ret = codec_process(dev);
 		if (ret)
 			return ret;
 
 		bytes_to_process -= codec->cpd.consumed;
 		processed += codec->cpd.consumed;
 		comp_update_buffer_consume(source, codec->cpd.consumed);
+		goto db_verify;
```

Comment:
not sure about this. You modified all `.process()` methods to only do initialisation if it hasn't been done yet. Before this patch the flow would continue and actually do processing below, now you skip it. Was this a bug? Is this intended?

Reply:
good catch fixed now